src/holofuel/RefundDeclinedOffers.js

```diff
@@ -32,17 +32,15 @@ function RefundDeclinedOffers ({
   const refundTransactions = useRefundTransactions()
   const declinedOffers = actionableTransactions.filter(transaction => ((transaction.status === STATUS.declined) && (transaction.type === TYPE.offer)))
 
-  const [refundedTransactionIds, setRefundedTransactionIds] = useState(false)
-  const toBeRefundedIds = difference(declinedOffers.map(offer => offer.id), refundedTransactionIds)
+  const [refundedTransactionIds, setRefundedTransactionIds] = useState([])
+  const toBeRefundedIds = difference(declinedOffers.map(declinedOffer => declinedOffer.id), refundedTransactionIds)
 
   useEffect(() => {
     if (!isEmpty(toBeRefundedIds)) {
-      refundTransactions(toBeRefundedIds)
-        .then(() => {
-          setRefundedTransactionIds(refundedTransactionIds => refundedTransactionIds.concat(toBeRefundedIds))
-        })
+      refundTransactions(declinedOffers)
+      setRefundedTransactionIds(refundedTransactionIds.concat(toBeRefundedIds))
```

Comment:
this change means we are updating the local state before we know whether the call succeeded. So if there is a zome error, the transactions will still be marked as refunded in the UI  so the UI will stop trying to refund them.

Reply:
True, but the promise.then was causing major issues, so I took it out for the time being.  Happy to look into why today.